flink-cdc-composer/src/test/java/org/apache/flink/cdc/composer/flink/FlinkPipelineTransformITCase.java

```diff
@@ -370,6 +371,99 @@ void testMultiTransformMissingProjection(ValuesDataSink.SinkApi sinkApi) throws
+        TransformDef nullProjection =
+                new TransformDef(
+                        "default_namespace.default_schema.mytable2",
+                        null,
+                        "age < 18",
+                        null,
+                        null,
+                        null,
+                        null,
+                        null);
+
+        TransformDef emptyProjection =
+                new TransformDef(
+                        "default_namespace.default_schema.mytable2",
+                        "",
+                        "age < 18",
+                        null,
+                        null,
+                        null,
+                        null,
+                        null);
+
+        TransformDef asteriskProjection =
+                new TransformDef(
+                        "default_namespace.default_schema.mytable2",
+                        "*",
+                        "age < 18",
+                        null,
+                        null,
+                        null,
+                        null,
+                        null);
+
+        List<TransformDef> transformDefList =
+                Arrays.asList(nullProjection, emptyProjection, asteriskProjection);
+
+        for (TransformDef transformDef : transformDefList) {
+            assertThatThrownBy(
+                            () ->
+                                    runGenericTransformTest(
+                                            sinkApi,
+                                            Arrays.asList(
+                                                    transformDef,
+                                                    new TransformDef(
+                                                            "default_namespace.default_schema.mytable2",
+                                                            // reference part column
+                                                            "id,UPPER(name) AS name",
+                                                            "age >= 18",
+                                                            null,
+                                                            null,
+                                                            null,
+                                                            null,
+                                                            null)),
+                                            Collections.emptyList()))
+                    .rootCause()
+                    .isExactlyInstanceOf(IllegalStateException.class)
+                    .hasMessage(
+                            "Unable to merge schema columns={`id` BIGINT,`name` VARCHAR(255),`age` TINYINT,`description` STRING}, primaryKeys=id, options=() "
+                                    + "and columns={`id` BIGINT,`name` STRING}, primaryKeys=id, options=() with different column counts.");
+        }
+
+        for (TransformDef transformDef : transformDefList) {
+            assertThatThrownBy(
```

Comment:
Seems this `assertThatTHrownBy` block still has no following assertions 🤕

Reply:
Thanks for the review!!
I modified the test case to be more rigorous, please help review again~
Deleted redundant assertThatThrownBy and I shouldn't use runGenericTransformTest in for loop, which will cause duplicate data events.